src/Lachain.Core/Network/FastSynchronizerBatch.cs/FastSynchronizerBatch.cs

```diff
@@ -6,24 +6,33 @@
 using Lachain.Storage.Trie;
 using Lachain.Storage.Repositories;
 using Lachain.Utility.Utils;
+using Lachain.Utility.Serialization;
+using System.Linq;
 
 namespace Lachain.Core.Network.FastSynchronizerBatch
 {
     
     public class FastSynchronizerBatch
     {
+        static string[] trieNames = new string[]
+        {
+                "Balances", "Contracts", "Storage", "Transactions", "Events", "Validators"
+        };
         private static readonly ILogger<FastSynchronizer> Logger = LoggerFactory.GetLoggerForClass<FastSynchronizer>();
         public static void StartSync(IStateManager stateManager,
                                      IRocksDbContext dbContext,
                                      ISnapshotIndexRepository snapshotIndexRepository,
                                      VersionFactory versionFactory,
                                      ulong blockNumber)
         {
+            if(Alldone(dbContext))
+            {
+                Console.WriteLine("Fast Sync was done previously\nReturning");
+                return;
+            }
             dbContext.Save(EntryPrefix.NodesDownloadedTillNow.BuildPrefix(), UInt64Utils.ToBytes(0));
             List<string> devnetNodes = new List<string>
```

Comment:
Why do we have hardcoded ips here?

Reply:
Yes, we can change that and take ips from config file, should I do it now? But in real scenario we need to be able find available peers by some other way.